### ISSUE TITLE
testing support

### ISSUE THREAD

**Issue by maks** [mishatron/flutter_auth0#8]
Thank you for such a nice package. I noticed there are no unit tests yet, I'd like to add some, but I think the main blocker to do so at the moment is that it's not possible to pass in `DioWrapper` as a constructor param to `Auth0Client`. 

I realise that most of the methods don't really have any business logic in them to test, but I think it would be useful in a few places, like `passwordGrant()` to test that `grant_type` is set correctly. 

Would you be open to changing the `Auth0Client` API to allow that?

**Comment by mishatron** [comment]
@maks Make a pull request with changes you want and I`ll check it. Thanks for contributing

**Comment by mishatron** [comment]
@maks Would you like to provide some tests?

**Comment by maks** [comment]
Yes sorry about the delay, I haven't had a chance to get back to working on the refactor to add tests. I'll try to get a PR ready shortly.
